frontend/src/metabase/query_builder/components/expressions/ExpressionEditorTextfield.jsx

```diff
@@ -86,18 +133,20 @@ export default class ExpressionEditorTextfield extends React.Component {
       const parserOptions = this._getParserOptions(newProps);
       const source = format(newProps.expression, parserOptions);
 
-      const { expression, compileError, syntaxTree } = this._processSource({
-        source,
-        ...this._getParserOptions(newProps),
-      });
-
+      const { expression, compileError, syntaxTree } =
+        source &&
+        source.length &&
```

Comment:
I don't think this guard will work since `const { foo } = undefined` will fail. Maybe that should be an early return at the start of `processSource`?

Reply:
yeah, that makes sense